src/variables.rs

```diff
@@ -125,6 +125,10 @@ impl<'a> Globals<'a> {
         Self(VariableMap::new())
     }
 
+    pub fn new_child(parent: &'a mut Globals<'a>) -> Self {
```

Comment:
The word "child" here threw me for a bit, but I see now that this is not merely a map from variables to values, but a _tree_ of such maps. I feel like this ought to be documented more clearly, but I guess that doesn't have to be in this PR.

Reply:
Good point! I can definitely add a comment. If you have a suggestion for a better name, I'm also open to that.

Reply:
For example, `new_nested` or `new_inherit` (or `new_inheriting`?).

Reply:
Hmm... `new_environment`, perhaps? Though I do like the explicitness of saying it's nested/inherited. One could also use the word "scope", though I feel like that's potentially overloaded already in this context.